### DISCUSSION TITLE
powershell command hangs with the base image from .NET Framework 4.8

### DISCUSSION THREAD

**Discussion by sajis997** [microsoft/dotnet-framework-docker#1153] [Q&A]
I have a Azure Windows VM (Windows Server 2022 DataCenter) with Docker CE installed with the following configuration:

Client:
 Version:           27.1.1
 API version:       1.46
 Go version:        go1.21.12
 Git commit:        6312585
 Built:             Tue Jul 23 19:57:57 2024
 OS/Arch:           windows/amd64
 Context:           desktop-windows

Server: Docker Desktop 4.33.1 (161083)
 Engine:
  Version:          27.1.1
  API version:      1.46 (minimum version 1.24)
  Go version:       go1.21.12
  Git commit:       cc13f95
  Built:            Tue Jul 23 19:56:46 2024
  OS/Arch:          windows/amd64
  Experimental:     false

I am trying to create a build pipeline for solution / project based on .NET Framework 4.8 and the build.dockerfile contains the following among others:

`FROM mcr.microsoft.com/dotnet/framework/sdk:4.8 AS build

ARG PACKAGE_VERSION

WORKDIR /V5FrameworkDiagnosticsBuild

COPY . ./

SHELL ["powershell", "-NoProfile", "-ExecutionPolicy", "Bypass", "-Command"]

RUN Write-Output 'Starting PowerShell version check...'; $PSVersionTable.PSVersion; Write-Output 'PowerShell version check completed.'

RUN Write-Output 'Listing directory contents...'; Get-ChildItem -Path .; Write-Output 'Directory listing completed.'

....................................`

While initiating the build.dockerfile from Github Actions workflow as follows:

      - name: Build inside a Docker Container
        run: docker build --build-arg PACKAGE_VERSION=${{ env.PACKAGE_VERSION }} --no-cache -t frameworkdiagcr01 -f .\build.dockerfile .
        shell: pwsh

It hangs at the poweshell command inside the dockerfile and eventually gives the following error message:

Step 7/11 : RUN Write-Output 'Starting PowerShell version check...'; $PSVersionTable.PSVersion; Write-Output 'PowerShell version check completed.'
 ---> Running in 33c09354f255
container 33c09354f2559185b7672ec891fba7349a423476db82a93083c1d28ddb20ffb4 encountered an error during hcs::System::Start: context deadline exceeded
Error: Process completed with exit code 1.

Any hint on why it happens the way it is and how to resolve the issue?

**Comment by lbussell** [comment]
Hi @sajis997, I was unable to reproduce this on my Windows Desktop using an identical version of Docker Desktop. Your Dockerfile build stops before using any .NET or VS tools, so I don't think the issue is with the .NET container image. Can you try to reproduce this using just the base Windows container image? For your image, it's `mcr.microsoft.com/dotnet/framework/runtime:4.8.1-20240709-windowsservercore-ltsc2022`:

https://github.com/microsoft/dotnet-framework-docker/blob/2fadd5ae013c62dd887a35cdb6d51147d7c1888b/src/sdk/4.8.1/windowsservercore-ltsc2022/Dockerfile#L3-L4

**Reply by sajis997** [reply]
Sorry, it did not help. It still hangs at the very first RUN echo command .The contents of the Dockerfile is as follows:

`ARG REPO=mcr.microsoft.com/dotnet/framework/runtime 
FROM $REPO:4.8.1-20240709-windowsservercore-ltsc2022 

ARG PACKAGE_VERSION

WORKDIR /V5FrameworkDiagnosticsBuild

COPY . ./

SHELL ["powershell", "-NoProfile", "-ExecutionPolicy", "Bypass", "-Command"]

RUN echo 'Starting PowerShell version check...' && \
    $PSVersionTable.PSVersion && \
    echo 'PowerShell version check completed.'

RUN echo 'Listing directory contents...' && \
    Get-ChildItem -Path . && \ 
    echo 'Directory listing completed.'

RUN echo "Restoring NuGet packages from NuGet.config file..." && \
    nuget restore V5FrameworkDiagnostics/V5FrameworkDiagnostics.sln -ConfigFile V5FrameworkDiagnostics/NuGet.Config && \
    echo "NuGet packages restored successfully."

RUN echo "Building the solution in release mode for x86 platform..." && \
    msbuild V5FrameworkDiagnostics/V5FrameworkDiagnostics.sln /p:Configuration=Release /p:Platform="x86" /p:OutputPath=./outX86 /m /nr:false /v:m && \
    echo "Solution built successfully."

RUN echo "Building V5DiagnosticsServiceInterface for Any CPU platform..." && \
    msbuild V5FrameworkDiagnostics/V5DiagnosticsServiceInterface/V5DiagnosticsServiceInterface.csproj /p:Configuration=Release /p:Platform="Any CPU" /p:OutputPath=./outAnyCpu /m /nr:false /v:m && \
    echo "V5DiagnosticsServiceInterface built for Any CPU platform."`
